### ISSUE TITLE
Add support for creating mementos from an aggregate root

### ISSUE THREAD

**Issue by einari** [Cratis/Arc#1601]
It should be possible to configure the aggregate for how often the memento should be generated and it should also be possible for the aggregate to manually create mementos.

The definition / shape of the memento should be defined by the aggregate root itself, typically by a type.

Depends on Cratis/Chronicle#1370

**Comment by woksin** [comment]
Triaging this for actionability. The direction is understood — configurable memento frequency, manual/on-demand memento creation, and an aggregate-defined memento shape — but there's no concrete API yet, and it references a dependency on Cratis/Chronicle#1370. @einari a few clarifications would let this be picked up:

- **Dependency status:** is Cratis/Chronicle#1370 resolved/available, or does this remain blocked on it? If blocked, this should probably be labeled/parked accordingly.
- **Frequency configuration:** where and how is the frequency expressed — per aggregate root type via attribute, fluent registration, or a global default with per-type override? Every N events, on a time interval, or both?
- **Manual creation API:** what's the intended call shape for creating a memento on demand (e.g. a method on the aggregate root, or via a service)?
- **Memento shape:** how does an aggregate define what goes into its memento — a dedicated method/type, or derived from the current state automatically?
- **Acceptance criteria:** a minimal example aggregate showing frequency config + a manual memento + the defined shape would make this verifiable.

Happy to implement once the API shape and the Chronicle#1370 dependency are settled.

---
_Generated by [Claude Code](https://claude.ai/code)_
